cmd/snap/cmd_debug_migrate.go

```diff
@@ -0,0 +1,80 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/strutil"
+)
+
+type cmdMigrateHome struct {
+	waitMixin
+
+	Positional struct {
+		Snaps []string `positional-arg-name:"<snap>" required:"1"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func init() {
+	addDebugCommand("migrate-home",
+		"Migrate snaps' directory to ~/Snap.",
+		"Migrate snaps' directory to ~/Snap.",
+		func() flags.Commander {
+			return &cmdMigrateHome{}
+		}, nil, nil)
+}
+
+func (x *cmdMigrateHome) Execute(args []string) error {
+	chgID, err := x.client.MigrateSnapHome(x.Positional.Snaps)
+	if err != nil {
+		msg, err := errorToCmdMessage("", "migrate-home", err, nil)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(Stderr, msg)
+		return nil
```

Comment:
Is it intentional that we "swallow" the error here? Shouldn't it be surfaced via an non-zero exit code?

Reply:
Yes, the `errorToCmdMessage` above determines if the error should be returned as an actual error or if we should just output something and exit with a zero code. If it's an actual error, it returns a non-nil error and we exit in line 53. If the error is an AlreadyInstalledError or a SnapNotInstalledError and the operation is a remove, then we just print something and exit with 0 here.